mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -2049,6 +2048,12 @@ void mlir::vector::populateSinkVectorBroadcastPatterns(
       patterns.getContext(), benefit);
 }
 
+void mlir::vector::populateReoderVectorTransposePatterns(
+    RewritePatternSet &patterns, PatternBenefit benefit) {
+  patterns.add<ReorderElementwiseOpsOnTranspose>(patterns.getContext(),
+                                                 benefit);
+}
```

Comment:
This only adds a single pattern. Should `populateSinkVectorBroadcastPatterns()` and `populateReoderVectorTransposePatterns()` be separate things (they're not used separately here).

Why not just `populateVectorReoderPatterns()`? 
If you need more control you could still add options to the populate call.